src/main/java/net/fortuna/ical4j/model/component/Observance.java

```diff
@@ -155,12 +155,21 @@ public final Date getLatestOnset(final Date date) {
 
         if (initialOnset == null) {
             try {
-                initialOnset = applyOffsetFrom(calculateOnset(((DtStart) getProperty(Property.DTSTART)).getDate()));
+                DtStart dtStart = (DtStart) getProperty(Property.DTSTART);
```

Comment:
One idea is also to add a getRequiredProperty method, which in turn will throw the ConstraintViolationException if property is missing.